src/assert.js

```diff
@@ -97,6 +97,26 @@ export function throws(blk, exp, msg) {
 	}
 }
 
+export async function rejects(blk, exp, msg) {
+	if (!msg && typeof exp === 'string') {
+		msg = exp; exp = null;
+	}
+
+	try {
+		await blk();
+		assert(false, false, true, 'rejects', false, 'Expected promise to reject', msg);
+	} catch (err) {
+		if (err instanceof Assertion) throw err;
+
+		if (typeof exp === 'function') {
+			assert(exp(err), false, true, 'rejects', false, 'Expected promise to reject matching exception', msg);
+		} else if (exp instanceof RegExp) {
+			const errorMessage = err instanceof Error ? err.message : err
+			assert(exp.test(errorMessage), false, true, 'rejects', false, `Expected promise to reject matching exception \`${String(exp)}\` pattern`, msg);
```

Comment:
```suggestion
			assert(exp.test(err.message), false, true, 'rejects', false, `Expected promise to reject matching exception \`${String(exp)}\` pattern`, msg);
```

Reply:
I know this (and suggestion below) are introducing a bug, but I'd prefer to (re-)fix it & the same issue in `throws` in the same PR, since these are mirrors of one another.

Reply:
Sure, I was a bit lazy and lumped that in there. It should really be a separate PR.